libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/api/ElasticsearchEntitlementChecker.java

```diff
@@ -420,4 +423,76 @@ public ElasticsearchEntitlementChecker(PolicyManager policyManager) {
     public void check$java_net_MulticastSocket$send(Class<?> callerClass, MulticastSocket that, DatagramPacket p, byte ttl) {
         policyManager.checkNetworkAccess(callerClass, NetworkEntitlement.CONNECT_ACTION | NetworkEntitlement.ACCEPT_ACTION);
     }
+
+    @Override
+    public void check$java_net_ServerSocket$(Class<?> callerClass, int port, int backlog, InetAddress bindAddr) {
+        policyManager.checkNetworkAccess(callerClass, NetworkEntitlement.LISTEN_ACTION);
+    }
+
+    @Override
+    public void check$java_net_ServerSocket$accept(Class<?> callerClass, ServerSocket that) {
+        policyManager.checkNetworkAccess(callerClass, NetworkEntitlement.ACCEPT_ACTION);
```

Comment:
Looks like you can work around a missing accept action entitlement by extending ServerSocket using a public constructor being granted only the listen action and overriding `accept` to call `implAccept` directly. Though, then an additional connect action is required to create the necessary Socket... Wondering if connect and listen are sufficient here?

Reply:
Maybe implAccept should be protected as well?

Reply:
> Maybe implAccept should be protected as well?

That's the solution I think; since ServerSocket is open to subclasses, implAccept is going to be part of the public surface. I'll add the check there too.